code/datums/components/suit_autodoc.dm

```diff
@@ -0,0 +1,340 @@
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	analyzer = new
+	if(!isnull(chem_cooldown))
+		src.chem_cooldown = chem_cooldown
+
+	src.burn_chems = burn_chems || default_burn_chems
+	src.oxy_chems = oxy_chems || default_oxy_chems
+	src.brute_chems = brute_chems || default_brute_chems
+	src.tox_chems = tox_chems || default_tox_chems
+	src.pain_chems = pain_chems || default_pain_chems
+	
+	if(!isnull(overdose_threshold_mod))
+		src.overdose_threshold_mod = overdose_threshold_mod
+
+/datum/component/suit_autodoc/Destroy(force, silent)
+	QDEL_NULL(analyzer)
+	QDEL_NULL(toggle_action)
+	QDEL_NULL(scan_action)
+	QDEL_NULL(configure_action)
+	wearer = null
+	return ..()
+
+/datum/component/suit_autodoc/RegisterWithParent()
+	. = ..()
+	toggle_action = new(parent)
+	scan_action = new(parent)
+	configure_action = new(parent)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, list(COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT, COMSIG_ITEM_DROPPED), .proc/dropped)
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED_TO_SLOT, .proc/equipped)
+	RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, .proc/action_toggle)
+	RegisterSignal(scan_action, COMSIG_ACTION_TRIGGER, .proc/scan_user)
+	RegisterSignal(configure_action, COMSIG_ACTION_TRIGGER, .proc/configure)
+
+/datum/component/suit_autodoc/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_PARENT_EXAMINE,
+		COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_EQUIPPED_TO_SLOT))
+	QDEL_NULL(toggle_action)
+	QDEL_NULL(scan_action)
+	QDEL_NULL(configure_action)
+
+/datum/component/suit_autodoc/proc/RegisterSignals(mob/user)
+	RegisterSignal(user, COMSIG_HUMAN_DAMAGE_TAKEN, .proc/damage_taken)
+
+/datum/component/suit_autodoc/proc/UnregisterSignals(mob/user)
+	UnregisterSignal(user, COMSIG_HUMAN_DAMAGE_TAKEN)
+
+/datum/component/suit_autodoc/proc/examine(datum/source, mob/user)
+	var/details
+	if(burn_nextuse > world.time)
+		details += "Its burn treatment injector is currently refilling. It will resupply in [(burn_nextuse - world.time) * 0.1] seconds.</br>"
+
+	if(brute_nextuse > world.time)
+		details += "Its trauma treatment injector is currently refilling. It will resupply in [(brute_nextuse - world.time) * 0.1] seconds.</br>"
+
+	if(oxy_nextuse > world.time)
+		details += "Its oxygenating injector is currently refilling. It will resupply in [(oxy_nextuse - world.time) * 0.1] seconds.</br>"
+
+	if(tox_nextuse > world.time)
+		details += "Its anti-toxin injector is currently refilling. It will resupply in [(tox_nextuse - world.time) * 0.1] seconds.</br>"
+
+	if(pain_nextuse > world.time)
+		details += "Its painkiller injector is currently refilling. It will resupply in [(pain_nextuse - world.time) * 0.1] seconds.</br>"
+
+	if(details)
+		to_chat(user, "<span class='danger'>[details]</span>")
+
+/datum/component/suit_autodoc/proc/dropped(datum/source, mob/user)
+	if(!iscarbon(user))
+		return
+	remove_actions()
```

Comment:
Passing `user` here would remove the need of `if(!wearer)` later.

Reply:
remove_actions() is called in places where user can't be passed without checking parent.loc and typecasting